packages/xstate-vue/test/UseMachine.vue

```diff
@@ -0,0 +1,71 @@
+<template>
+  <div>
+    Hallo
+    <button v-if="current.matches('idle')" @click="send('FETCH')">Fetch</button>
+    <div v-else-if="current.matches('loading')">Loading...</div>
+    <div v-else-if="current.matches('success')">
+      Success! Data:
+      <div data-testid="data">{{ current.context.data }}</div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { useMachine } from '../src/useMachine';
+import { Machine, assign, Interpreter, spawn, doneInvoke, State } from 'xstate';
+import { watch } from '@vue/composition-api';
+
+const context = {
+  data: undefined
+};
+const fetchMachine = Machine<typeof context, any>({
+  id: 'fetch',
+  initial: 'idle',
+  context,
+  states: {
+    idle: {
+      on: { FETCH: 'loading' }
+    },
+    loading: {
+      invoke: {
+        src: 'fetchData',
+        onDone: {
+          target: 'success',
+          actions: assign({
+            data: (_, e) => e.data
+          }),
+          cond: (_, e) => e.data.length
+        }
+      }
+    },
+    success: {
+      type: 'final'
+    }
+  }
+});
+
+export default {
+  props: ['persistedState'],
+  setup({ persistedState }): { persistedState: State<any, any> } {
+    const onFetch = () =>
+      new Promise(res => setTimeout(() => res('some data'), 50));
+
+    const { current, send, service } = useMachine(fetchMachine, {
+      services: {
+        fetchData: onFetch
+      },
+      state: persistedState
+    });
+    watch(() =>
+      current.value.actions.forEach(action => {
+        if (action.type === 'load') {
```

Comment:
wouldnt this already be handled by the interpreter in `useMachine`?

Reply:
I tried to follow the equivalent [react hook test](https://github.com/davidkpiano/xstate/blob/07f390f450219f7b25a024e99e35e3d534489266/packages/xstate-react/test/useFSM.test.tsx#L46) while implementing that test, so maybe? :)

Reply:
Ok - this is actually for `useMachine` with `@xstate/fsm` so it doesn't have action execution builtin to its interpreter 👍